ui/packages/shared/profile/src/ProfileIcicleGraph/IcicleGraphArrow/index.tsx

```diff
@@ -174,8 +174,23 @@ export const IcicleGraphArrow = memo(function IcicleGraphArrow({
   };
 
   const hideBinary = (binaryToRemove: string): void => {
-    const newMappingsList = mappingsList.filter(mapping => mapping !== binaryToRemove);
+    // second/subsequent time filtering out a binary i.e. a binary has already been filtered out
+    // and we want to filter out another binary, we need to discover what the first one was
+    // and then add more as needed
+    if (binaryFrameFilter !== undefined && binaryFrameFilter.length > 0) {
+      const binariesToExclude = mappingsList.filter(
```

Comment:
why filter the original list again? `binaryFrameFilter` is already the "current" list of binaries to show if it's non-empty, so just filter that, no?